src/coreclr/vm/method.cpp

```diff
@@ -1208,7 +1208,7 @@ COR_ILMETHOD* MethodDesc::GetILHeader()
     // Always pickup overrides like reflection emit, EnC, etc.
```

Comment:
Notice that this method has a `PRECONDITION(IsIL())`. It means that it is not expected to be called on `IsNDirect` method in the first place.

I think the correct fix is to add `if (IsIL())` check around the whole CodeVersionManager related block in `MethodDesc::EnumMemoryRegions`.